aws/resource_aws_lambda_event_source_mapping.go

```diff
@@ -156,6 +163,66 @@ func resourceAwsLambdaEventSourceMapping() *schema.Resource {
 					},
 				},
 			},
+			/*
```

Comment:
The suggested format for this from @ewbankkit from issue #16785 was:

```
  self_managed_event_source {
    endpoints = {
      KAFKA_BOOTSTRAP_SERVERS = ["abc3.xyz.com:9092", "abc2.xyz.com:9092"]
    }
  }
```

I was unable to make this work.  The parsing of the terraform file would panic as soon as I tried to make
the map value into a list.  I saw some references that TypeMap could only have primitive elements.  If that
isn't the case, I would love to know how to make this work.

Reply:
@delbaeth Yes, that is a limitation in the current Terraform Provider type system.
The underlying AWS API type `map[string][]*string` is rarely used and the only place I can find where we are using it currently is in the `aws_budgets_budget` resource's `cost_filters` attribute. There it looks like the list of strings is handled as a comma-separated single string, so the equivalent would be

```hcl
self_managed_event_source {
  endpoints = {
    KAFKA_BOOTSTRAP_SERVERS = "abc3.xyz.com:9092,abc2.xyz.com:9092"
  }
}
```

and the resource handling code will have to do the `strings.Split` and `strings.Join`.
What do you think?

Reply:
I implemented it that way as well.  Happy to go back to it.  The comma separated list is standard for bootstrap brokers
anyway.

The reason I switched was the API seemed to be intended for generic use
and might at some point be used for something other than kafka.  In that case
the choice of using a comma as the separator might not work as well.
If I needed special handling for the kafka case I figured I might as well
just define it explicitly.

Anyway, none of the implementations seemed great.  I will go back to the comma separated list.